src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -427,5 +528,194 @@ private ImmutableArray<BoundExpression> BindInterpolatedStringParts(BoundUnconve
             parameterNamesAndLocationsBuilder.Free();
             return (builderAppendCalls.ToImmutableAndFree(), builderPatternExpectsBool ?? false);
         }
+
+        private BoundExpression BindInterpolatedStringHandlerInMemberCall(
+            BoundUnconvertedInterpolatedString unconvertedString,
+            ArrayBuilder<BoundExpression> arguments,
+            ImmutableArray<ParameterSymbol> parameters,
+            ref MemberAnalysisResult memberAnalysisResult,
+            int interpolatedStringArgNum,
+            TypeSymbol? receiverType,
+            RefKind? receiverRefKind,
+            BindingDiagnosticBag diagnostics)
+        {
+            var interpolatedStringConversion = memberAnalysisResult.ConversionForArg(interpolatedStringArgNum);
+            Debug.Assert(interpolatedStringConversion.IsInterpolatedStringHandler);
+            var interpolatedStringParameter = GetCorrespondingParameter(ref memberAnalysisResult, parameters, interpolatedStringArgNum);
+            Debug.Assert(interpolatedStringParameter.Type is NamedTypeSymbol { IsInterpolatedStringHandlerType: true });
+
+            if (interpolatedStringParameter.HasInterpolatedStringHandlerArgumentError)
+            {
+                // The InterpolatedStringHandlerArgumentAttribute applied to parameter '{0}' is malformed and cannot be interpreted. Construct an instance of '{1}' manually.
+                diagnostics.Add(ErrorCode.ERR_InterpolatedStringHandlerArgumentAttributeMalformed, unconvertedString.Syntax.Location, interpolatedStringParameter, interpolatedStringParameter.Type);
+                return CreateConversion(
+                    unconvertedString.Syntax,
+                    unconvertedString,
+                    interpolatedStringConversion,
+                    isCast: false,
+                    conversionGroupOpt: null,
+                    wasCompilerGenerated: false,
+                    interpolatedStringParameter.Type,
+                    diagnostics,
+                    hasErrors: true);
+            }
+
+            var handlerParameterIndexes = interpolatedStringParameter.InterpolatedStringHandlerArgumentIndexes;
```

Comment:
nit: not related to this PR, consider updating the xml doc on `InterpolatedStringHandlerArgumentIndexes` with references to the special values such as `BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter` and others